### ISSUE TITLE
Filter sets that are easy to activate/deactivate

### ISSUE THREAD

**Issue by smatthie** [mastodon/mastodon#20483]
### Pitch

The user should be able to create filter sets (with several individual filters) and it should be easy to turn them on and off from the startpage. This will give users more power to decide what kind of topics they would like to see at the moment (depending on their mood).

For example, the user is generally interested in politics, but sometimes they might be stressed and can't handle it and only want to see the nice cat pictures in their timeline. So they can create a filter set "politics" that contains, e.g., filters for  "politics", "government", "parliament", "[Name of politician A]", "[Name of politician B]" and so in, and also for all content from certain accounts (e.g if the user is following a politician).

On the timeline there is a switch so they can turn the filter set "politics" on and off simultaneously when they feel like it, and all the individual filters get turned on or off accordingly.

### Motivation

Currently it is quite inconvenient to set up filters, in particular for whole topic areas (rather than individual words), and you can't easily turn them on and off.

Partly this is motivated by the discussions around CWs. Without wanting to rehash this, some people use CWs in the way I described above to filter out any topic they find distracting at the moment. Consequently they ask others to put CW all sorts of topics, and much disagreement ensues. 

The reason why people now use CWs (rather than filters) is that CWs can be made visible/invisible with one click. However, CWs put the onus on the author of the post to label them correctly. While this may be desirable, in practice users don't agree what should be labelled, and it is hard for authors to know what readers may or may not want to see at any moment. And some authors simply disagree with the principle. So readers are frustrated as authors don't use the CWs they want.

Basically, CWs are controlled by the author, filters are controlled by the reader.

So for the use cases outlined above (using CWs for filtering any topic rather than warn of traumatic content) it makes much more sense to use filters rather than CWs, but this is hampered by the fact that filters need a lot of clicks to be set up and can't be turned on and off.

Therefore the proposal is meant to make filters easier to use, as filter sets enable the reader to have much finer control over their timeline and not depend on the author to label things for them.

**Comment by ClearlyClaire** [comment]
Which version of Mastodon is your instance running? v4.0.0 will have filters grouped by category, and a whole category can share an expiration date for instance. This doesn't completely satisfy your request though as:
- there is no way to outright disable a category
- it's only modifiable in the settings

**Comment by smatthie** [comment]
Thank you, I'm still on 3.5.3. but that sounds promising.

A key feature of my request is the ability to turn them on and off very easily (from the start page), so I'd like to keep this request open, if that's ok?

**Comment by ClearlyClaire** [comment]
> A key feature of my request is the ability to turn them on and off very easily (from the start page), so I'd like to keep this request open, if that's ok?

Yes! I would have closed it otherwise!

**Comment by 31SFX4** [comment]
I have now tested the new filter system in 4.0 and indeed is a huge improvement that addresses many of the issues that motivated me to propose this, with combining filters under one heading and also the option to show a hint in the timeline. 

However, I would still like to be able to enable and disable filter sets globally without having to delete and recreate them, so I will keep this open and make a note in the original proposal.

**Comment by Aligorith** [comment]
Glad I stumbled across this issue just before submitting a new one.

The "Filter this post" option does indeed make things a lot easier than having to bring up the Preferences to do it each time. A few thoughts on ways this could be betters still:
* I was originally going to recommend putting an "Add / Edit" Filters button as part of the "Show Settings" pane at the top of each timeline / thread. Even with the "Filter this post" option, I think it would still be good to have it.

* Making the dialog "add filter" dialog modal when trying to create filters makes the workflow a bit clunky still, as it means I can't easily select the annoying terms from the original post to include in the filter.
